### ISSUE TITLE
Error using python args on pex interpreter

### ISSUE THREAD

**Issue by DaviAmorim29** [pantsbuild/pex#1737]
I want to use example a -u python args on pex, how can i do that ?

if i use:
ex:
pex requests -u myscript.py
r: pex: error: unrecognized arguments: -u myscript.py

I don't see nothing on the documentation talking about this

**Comment by jsirois** [comment]
It is true that a PEX interpreter does not support all Python interpreter args directly; just `-m` and `-c` currently.
https://github.com/pantsbuild/pex/blob/c851f4c138622fb008c0c639d8e1344b83767b7c/pex/pex.py#L594-L620

1st, pretending a PEX interpreter did support `-u`, you'd do things just a bit differently than in your example. In your example you are not using a PEX interpreter, but you're calling the Pex CLI tool instead. You're effectively creating a throwaway PEX interpreter since you do not specify a `-o / --output` file for the PEX. To pass arguments to an ephemeral PEX like this, you use `--` to separate them from the Pex CLI args. So you'd use:
```
$ pex requests -- -u myscript.py
Could not open -u in the environment [/tmp/tmpk8pxc7ns]: [Errno 2] No such file or directory: '-u'
```

And if PEX interpreters supported `-u`, that would work. Similarly, you could 1st build a PEX requests interpreter:
```
$ pex requests -o requests.pex
```
And then run your script with it:
```
$ ./requests.pex -u myscript.py 
Could not open -u in the environment [/home/jsirois/Downloads/./requests.pex]: [Errno 2] No such file or directory: '-u'
```

PYTHON* env vars work though; so given this script:
```python
import os
import sys
import time

import requests


response = requests.get("https://example.org", stream=True)
for line in response.iter_lines(decode_unicode=True):
    sys.stdout.write(line)
    time.sleep(0.1)
```

You'll see choppy output like you'd expect from this:
```
$ PYTHONUNBUFFERED=x ./requests.pex myscript.py
```

**Comment by jsirois** [comment]
@DaviAmorim29 hopefully this answers your question and provides a work-around. I'm going to leave this issue open though because it seems like there is more Pex could be doing here to be a bit more transparent and act more exactly like a Python interpreter (+ deps).

**Comment by DaviAmorim29** [comment]
Thanks for the help men !! Very good

**Comment by jsirois** [comment]
I've looked at this @DaviAmorim29 and there is no reason PEXes cannot detect arguments coming before the `-c <code> | -m <module> | - | <file> | <dir>` terminating arguments to the Python interpreter in this case and re-exec to get those arguments handled by Python. Pex knows which interpreter its running under and it can knows the command line flags of all Pythons it supports (CPython 2.7 and 3.5+ as well as PyPy2 and PyPy3). Although you seem to be happy with the workaround, I've opened #1745 to track making PEX interpreters behave _exactly_ like the underlying Python interpreter + deps.
